File: Searching/Exponential/Exponential.java
public class Exponential {

    // Binary Search used by Exponential Search
    public static int binarySearch(int[] arr, int target, int low, int high) {
        while (low <= high) {
            int mid = (low + high) / 2;
            if (arr[mid] == target) {
                return mid;
            } else if (arr[mid] < target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }
        return -1;
    }

    // Exponential Search
    public static int exponentialSearch(int[] arr, int target) {
        if (arr.length == 0) return -1;

        if (arr[0] == target) return 0;

        int i = 1;
        while (i < arr.length && arr[i] <= target) {
            i *= 2;
        }

        return binarySearch(arr, target, i / 2, Math.min(i, arr.length - 1));
    }

    public static void main(String[] args) {
        int[] arr = {2, 3, 4, 10, 40, 60, 80};
        int target = 10;

        int result = exponentialSearch(arr, target);

        if (result != -1) {
            System.out.println("Element found at index: " + result);
        } else {
            System.out.println("Element not found.");
        }
    }
}

File: Searching/Exponential/Exponential.py
def binary_search(arr, target, low, high):
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1


def exponential_search(arr, target):
    if not arr:
        return -1

    if arr[0] == target:
        return 0

    i = 1
    while i < len(arr) and arr[i] <= target:
        i *= 2

    # Do binary search between i/2 and min(i, len(arr)-1)
    return binary_search(arr, target, i // 2, min(i, len(arr) - 1))
